src/main/scala/com/yotpo/metorikku/output/writers/elasticsearch/ElasticsearchOutputWriter.scala

```diff
@@ -0,0 +1,46 @@
+package com.yotpo.metorikku.output.writers.elasticsearch
+
+import com.yotpo.metorikku.configuration.job.output.Elasticsearch
+import com.yotpo.metorikku.output.Writer
+import org.apache.log4j.LogManager
+import org.apache.spark.sql.{DataFrame, SaveMode}
+
+
+class ElasticsearchOutputWriter(props: Map[String, String], elasticsearchOutputConf: Option[Elasticsearch]) extends Writer {
+
+  case class ElasticsearchOutputProperties(saveMode: SaveMode, resource: String, mappingId: String, writeOperation: String,
+                                           port: Int, indexAutoCreate: Boolean, nodesWanOnly: Boolean)
+
+  val log = LogManager.getLogger(this.getClass)
+
+  val saveMode: SaveMode = SaveMode.valueOf(props.getOrElse("saveMode", "Overwrite"))
+  val port: Int = props.getOrElse("port", "9200").toString.toInt
+  val indexAutoCreate: Boolean = props.getOrElse("indexAutoCreate", true).toString.toBoolean
+  val nodesWanOnly: Boolean = props.getOrElse("nodesWanOnly", true).toString.toBoolean
+
+  val dbOptions = ElasticsearchOutputProperties(saveMode,
+                                                props("resource"),
+                                                props.getOrElse("mappingId", ""),
+                                                props.getOrElse("writeOperation", ""),
+                                                port,
+                                                indexAutoCreate,
+                                                nodesWanOnly)
+
+  override def write(dataFrame: DataFrame): Unit = {
+    val writer = dataFrame.write.format("org.elasticsearch.spark.sql")
+      .option("es.port", dbOptions.port)
+      .option("es.nodes.wan.only", dbOptions.nodesWanOnly)
+      .mode(dbOptions.saveMode)
+
+    if (!dbOptions.mappingId.isEmpty()) {
+      writer.option("es.mapping.id", dbOptions.mappingId)
+    }
+    if (!dbOptions.writeOperation.isEmpty()) {
+      if (dbOptions.saveMode != SaveMode.Append) {
+        log.warn("When using 'upsert'/'update' writeOperation, saveMode must be 'Append'")
+      }
+      writer.option("es.write.operation", dbOptions.writeOperation)
+    }
+    writer.save("%s/_doc".format(dbOptions.resource))
```

Comment:
I think you should allow sending any extra option they like on top of what you offer.

Reply:
You need to support sending the type (not only _doc) for previous versions of elasticsearch